### PR TITLE
fix: graphql test cases moved to server side tests from client side tests folder

### LINKED ISSUE #13367: [Feature]: Add Test Cases for GraphQL
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Need to add graphql test cases for the automation testing

### Why should this be worked on?

Automation Testing

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

Graphql Tests Added for creation, execution without pagination, execution with limit based pagination

Fixes #13367

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Running Test cases

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts [hunk 1]
```diff
@@ -38,141 +42,132 @@ const GRAPHQL_LIMIT_DATA = [
...
+    });
   });
 
   it("3. Pagination for limit based should work with offset", function() {
     /* Create an API */
-    cy.NavigateToActiveDSQueryPane(datasourceName);
-    cy.get(apiEditorLocators.graphqlQueryEditor)
+    dataSources.NavigateFromActiveDS(datasourceName, true);
+
+    cy.get(dataSources._graphqlQueryEditor)
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts:121]: datacources.EnterQuery() - can be checked if it works here.. No direct cy statements in ts specs please.

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts [hunk 2]
```diff
@@ -38,141 +42,132 @@ const GRAPHQL_LIMIT_DATA = [
...
 
     // Select Limit base Pagination
-    cy.get(`${apiEditorLocators.apiPaginationTab} label`)
-      .eq(1)
-      .click({ force: true });
+    apiPage.SelectPaginationTypeViaIndex(1);
 
     // Select Limit Variable from dropdown
-    cy.get(apiEditorLocators.graphqlPagination.limitVariable).click({
+    cy.get(dataSources._graphqlPagination.limitVariable).click({
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts:135]: Use     this.agHelper.GetNClick()- here

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts [hunk 3]
```diff
@@ -38,141 +42,132 @@ const GRAPHQL_LIMIT_DATA = [
...
 
     // Set the Limit Value as 1
-    cy.get(apiEditorLocators.graphqlPagination.offsetValue)
+    cy.get(dataSources._graphqlPagination.offsetValue)
       .first()
       .focus()
       .type("1");
 
-    cy.wait(500);
+    cy.wait(1000);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts:162]: Use this.agHelper.Sleep() here

**Review by Aishwarya-U-R — CHANGES_REQUESTED** [review]

File: app/client/cypress/support/Pages/DataSources.ts [hunk 4]
```diff
@@ -387,4 +406,35 @@ export class DataSources {
     });
     this.agHelper.AssertAutoSave();
   }
+
+  public CreateGraphqlDatasource(datasourceName: string) {
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/support/Pages/DataSources.ts:425]: Added the function, changed the logic

File: app/client/cypress/support/Pages/DataSources.ts [hunk 5]
```diff
@@ -66,6 +66,25 @@ export class DataSources {
...
+    offsetValue: ".t--apiFormPaginationOffsetValue .CodeMirror textarea",
+    prevLimitVariable: ".t--apiFormPaginationPrevLimitVariable",
+    prevLimitValue: ".t--apiFormPaginationPrevLimitValue .CodeMirror textarea",
+    prevCursorVariable: ".t--apiFormPaginationPrevCursorVariable",
+    prevCursorValue:
+      ".t--apiFormPaginationPrevCursorValue .CodeMirror textarea",
+    nextLimitVariable: ".t--apiFormPaginationNextLimitVariable",
+    nextLimitValue: ".t--apiFormPaginationNextLimitValue .CodeMirror textarea",
+    nextCursorVariable: ".t--apiFormPaginationNextCursorVariable",
+    nextCursorValue:
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/support/Pages/DataSources.ts]: Changed the variable names.

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_creation_spec.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_creation_spec.ts]: Done

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts (hunk 2 above)
Comment (AmanAgarwal041) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts]: Moved the logic to `datasources` file. As all the locators were different so did not use `this.agHelper.GetNClick`

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts (hunk 3 above)
Comment (AmanAgarwal041) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/Datasources/GraphQL_execution_spec.ts]: done

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving cypress tests!

**Comment by AmanAgarwal041** [comment]
> Approving cypress tests!

Adding `Test Plan Approved` label

**Pull request merged**